File: examples/javascript/ds1307.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2014 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

// Load RTC Clock module for Grove - RTC clock
var ds1307 = require('jsupm_ds1307');
// load this on i2c bus 0
var myRTCClockObj = new ds1307.DS1307(0);

// always do this first
console.log("Loading the current time... ");

var result = myRTCClockObj.loadTime();
if (!result)
{
	console.log("myRTCClockObj.loadTime() failed.");
	process.exit(1);
}

printTime(myRTCClockObj);

// set the year as an example
console.log("setting the year to 50");
myRTCClockObj.year = 50;
myRTCClockObj.setTime();

// reload the time and print it
myRTCClockObj.loadTime();
printTime(myRTCClockObj);

function printTime(RTCObj)
{
	var timeStr = "The time is: " + 
        RTCObj.month + "/" + RTCObj.dayOfMonth + "/" + RTCObj.year + " " +
        RTCObj.hours + ":" + RTCObj.minutes + ":" + RTCObj.seconds;

	if (RTCObj.amPmMode)
		timeStr += (RTCObj.pm ? " PM " : " AM ");

	console.log(timeStr);

	console.log("Clock is in " + 
                (RTCObj.amPmMode ? "AM/PM mode" : "24hr mode"));
}

File: examples/c++/ds1307.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2014 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <string>

#include "ds1307.hpp"

using namespace std;

void
printTime(upm::DS1307& rtc)
{
    cout << "The time is: " << rtc.month << "/" << rtc.dayOfMonth << "/" << rtc.year << " "
         << rtc.hours << ":" << rtc.minutes << ":" << rtc.seconds;

    if (rtc.amPmMode)
        cout << ((rtc.pm) ? " PM " : " AM ");

    cout << endl;

    cout << "Clock is in " << ((rtc.amPmMode) ? "AM/PM mode" : "24hr mode") << endl;
}

int
main(int argc, char** argv)
{
    //! [Interesting]
    // Instantiate a DS1037 on I2C bus 0
    upm::DS1307 rtc(0);

    // always do this first
    cout << "Loading the current time... " << endl;
    if (!rtc.loadTime()) {
        cerr << "rtc.loadTime() failed." << endl;
        return 0;
    }

    printTime(rtc);

    // set the year as an example
    cout << "setting the year to 50" << endl;
    rtc.year = 50;

    rtc.setTime();

    // reload the time and print it
    rtc.loadTime();
    printTime(rtc);

    //! [Interesting]

    return 0;
}
